packages/@react-spectrum/utils/src/index.ts

```diff
@@ -29,8 +29,8 @@ export {
   getResponsiveProp
 } from './styleProps';
 export {useSlotProps, cssModuleToSlots, SlotProvider, ClearSlots} from './Slots';
-export {useHasChild} from './useHasChild';
-export {useIsMobileDevice} from './useIsMobileDevice';
+export {useHasChild} from '@react-aria/utils';
+export {useIsMobileDevice} from '@react-aria/utils';
```

Comment:
I was thinking about it more, we don't need to move either of these

useIsMobileDevice is no longer used

and in place of useHasChild, we can actually find out if the child exists by providing a slotCallbackSymbol, if it's called, then we know we have that child

can see https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/src/ComboBox.tsx#L128
we'll need to be careful not to put the props into state, because that may cause extra re-renders, just if that callback is invoked, we should set hasInput to to true

Reply:
There's also `useSlot` in react-aria-components utils.tsx, which returns a callback ref to register when a slot is present.